packages/api-clients/open-api/agreementApi.yml

```
@@ -1270,6 +1273,37 @@ components:
       required:
         - results
         - totalCount
+    Stamps:
+      type: object
+      additionalProperties: false
+      properties:
+        submission:
+          $ref: "#/components/schemas/StampInfo"
+        activation:
+          $ref: "#/components/schemas/StampInfo"
+        rejection:
+          $ref: "#/components/schemas/StampInfo"
+        suspensionByProducer:
+          $ref: "#/components/schemas/StampInfo"
+        suspensionByConsumer:
+          $ref: "#/components/schemas/StampInfo"
+        upgrade:
+          $ref: "#/components/schemas/StampInfo"
+        archiving:
+          $ref: "#/components/schemas/StampInfo"
+    StampInfo:
```

Just a minor comment: we could simply call it `Stamp`
```suggestion
    Stamp:
```